File: integration-tests/iac/lambdas/python/sqs_producer.py
import json
import os
import boto3

sqs = boto3.client('sqs')


def handler(event, context):
    queue_url = os.environ['QUEUE_URL']

    message = {
        'message': 'Hello from SQS producer!',
        'request_id': context.aws_request_id,
    }

    response = sqs.send_message(
        QueueUrl=queue_url,
        MessageBody=json.dumps(message),
    )

    print(f"Sent message to SQS: {response['MessageId']}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'message_id': response['MessageId'],
        }),
    }

File: integration-tests/iac/lambdas/node/sqs_producer.mjs
import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs';

const sqs = new SQSClient();

export async function handler(event, context) {
    const queueUrl = process.env.QUEUE_URL;

    const message = {
        message: 'Hello from SQS producer!',
        request_id: context.awsRequestId,
    };

    const response = await sqs.send(new SendMessageCommand({
        QueueUrl: queueUrl,
        MessageBody: JSON.stringify(message),
    }));

    console.log(`Sent message to SQS: ${response.MessageId}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            message_id: response.MessageId,
        }),
    };
}
